src/osd/PrimaryLogPG.cc

```diff
@@ -418,6 +418,8 @@ void PrimaryLogPG::on_local_recover(
       ceph_assert(recovering.count(obc->obs.oi.soid));
       recovering[obc->obs.oi.soid] = obc;
       obc->obs.oi = recovery_info.oi;  // may have been updated above
+      if (!recovery_info.soid.is_snap())
```

Comment:
There is a `has_snapset` helper, which I think would make the code more readable. Would you mind switching to use that?

Reply:
my pleasure, thanks remind. :D